komodo/lint_maturity.py

```diff
@@ -0,0 +1,255 @@
+        )
+        system_warning_msg += msg_release_exception(release_basename, release_version)
+
+        if release_version == "invalid":
+            system_exit_msg += (
+                release_basename + " is incompatible with version name.\n"
+            )
+        else:
+            packages_dict = read_yaml_file(file_path=file_to_lint)
+            dict_tag_maturity = get_packages_info(
+                packages_dict, tag_exceptions["package"]
+            )
+            count_tag_invalid = count_invalid_tags(
+                dict_tag_maturity, _INVALID_TAGS[release_version]
+            )
+
+            system_warning_msg += msg_packages_exception(
+                release_basename, dict_tag_maturity
+            )
+
+            if count_tag_invalid > 0:
+                packages_msg = msg_packages_invalid(
+                    release_basename,
+                    release_version,
+                    count_tag_invalid,
+                    dict_tag_maturity,
+                )
+
+                if release_version != "exception":
+                    system_exit_msg += packages_msg
+                else:
+                    system_warning_msg += packages_msg
+
+    print_warning_message(system_warning_msg)
+    print_system_exit_message(system_exit_msg)
+
+
+def get_files_to_lint(args):
+
+    if args["release_folder"] is None:
+        files_to_lint = [args["release_file"]]
+    else:
+        files_to_lint = [
+            os.path.join(args["release_folder"], file_in_dir)
+            for file_in_dir in os.listdir(args["release_folder"])
+        ]
+
+    return files_to_lint
+
+
+def define_tag_exceptions(tag_exception_arg):
+
+    if os.path.isfile(tag_exception_arg[0]):
+        tag_exceptions = read_yaml_file(file_path=tag_exception_arg[0])
+    elif tag_exception_arg[0] == "":
+        tag_exceptions = {"release": [], "package": []}
+    else:
+        exit_msg = str(tag_exception_arg) + " is not a valid file."
+        raise SystemExit(exit_msg)
+
+    return tag_exceptions
+
+
+def get_parser():
+
+    parser = argparse.ArgumentParser(description=("Lint the maturity of packages."))
+    parser.add_argument(
+        "--tag_exceptions",
+        nargs="+",
+        default=[""],
+        help="One file with exceptions for (a) release tag and (b) packages tag.",
+    )
+
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument(
+        "--release_file",
+        type=lambda arg: arg
+        if os.path.isfile(arg)
+        else parser.error("{} is not a valid file".format(arg)),
+        help="Release file",
+    )
+    group.add_argument(
+        "--release_folder",
+        type=lambda arg: arg
+        if os.path.isdir(arg)
+        else parser.error("{} is not a valid directory".format(arg)),
+        help="File with all package tags named as release version.",
+    )
+
+    return parser
+
+
+def main():
+
+    parser = get_parser()
+    args = vars(parser.parse_args())
```

Comment:
Why do you want to run it through `vars` by the way? I would think it would be better not to do that.

Reply:
Thanks! 
It was a (perhaps non-elegant) way to answer a previous review comment, to test the function ´main()´.
I tried to break the function further by defining get_files_to_lint, and avoided testing large strings from ´main´ outcomes (which turned to be difficulty because of /t and /n).
To test get_files_to_lint, it appeared that having the variable arg as a dictionary was more feasible compared to creating a parsed object.
Why is it not good? Would you have a suggestion to create and test a function that call a parser object?
Many thanks!

Reply:
Sure, the best way would be to use a `mock`, for example like this (also see my unit test of `main` example):

```
args_mock = mock.Mock()
args_mock.release_folder = "some/folder"
args_mock.release_file = "some_fname"
```

Note that you could bypass this whole problem by not passing in `args` to `get_files_to_lint`, which might actually be preferable, as you want a function to work on as basic types as possible, and not "know" too much.

Reply:
My preference would be just that, to pass two strings.